### ISSUE TITLE
Unexpected warning for equivalent notation

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#41884]
**Version Used**: 16.5 Preview 3

**Steps to Reproduce**:

```csharp
#nullable enable

using System.Diagnostics.CodeAnalysis;

public class Base<T> {
  public virtual bool Method([MaybeNullWhen(false)] out T value) => throw null!;
}

public class Derived<T> : Base<T>
  where T : class {
  public override bool Method([NotNullWhen(true)] out T? value) => throw null!;
}
```

**Expected Behavior**:

No diagnostics.

**Actual Behavior**:

warning CS8610: Nullability of reference types in type of parameter 'value' doesn't match overridden member.

**Comment by sharwell** [comment]
Examples in Roslyn:

https://github.com/dotnet/roslyn/blob/24e61468355fb43e4b22a6e9f39816ee3e773f76/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ValuesSources/WeaklyCachedValueSource.cs#L34-L36

https://github.com/dotnet/roslyn/blob/24e61468355fb43e4b22a6e9f39816ee3e773f76/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ValuesSources/WeaklyCachedRecoverableValueSource.cs#L67-L69

**Comment by jcouv** [comment]
I believe @RikkiGibson fixed this (better accounting of nullability attributes in comparing signatures for overrides/implementations). I verified there is no warning now ([sharplab](https://sharplab.io/#v2:CYLg1APgxAdgrgGwQQwEYIKYAIMzZgWAChiABABi1IEYA6AEQEtkBzGAewGcAXRgY060Awu2AYAgngQBPTo04BuYmQDMVAExYAQsk4YAPABUAfFgDexLFTUA3RgCducZAiyp27VwFkM3ABaiABQA2l7I0qgYAHKICADqfriBAGYuegCUALpY7HDcWIZYNi5wGOlYALym/vbsAO5Y8EgAhEpEAL7KRKRqpJr0GPaMNhjARqYg2roGJpZYdYn22IWTfeZzPTkj9kNibh7evgHAIVHs3DFICUnc9qVZOXkFAPxFJWWV1X61DU0IrcR2kA==)).
I'll go ahead and close as fixed. Thanks
